### DISCUSSION TITLE
start stream from recordings

### DISCUSSION THREAD

**Discussion by Clemens-E** [bluenviron/mediamtx#2763] [Questions]
### Question

Hi, first of all, really great software with super low resource usage. Thrilled to see something like this.
I'm streaming from an ip camera and I enabled saving files in 5 Minute chunks in the fmp4 format. 
I'm wondering if there is any native functionality of mediamtx to specify you want to start the stream in the past and mediamtx automatically finds the needed position in the recordings and starts streaming from there.

That would be fantastic. Otherwise, I guess I have to write a server that uses the requested timeframe to find the correct chunk, then calculate at what rough position it has to start streaming and go from there. I'm just not sure how I can deal with the case that the chunk ends, and I would have to seamlessly go to the next one, but I guess I can just start a second request with the new timeframe.

**Comment by aler9** [comment]
Hello, this is planned and tracked here #2452

In the meanwhile you can write the server by your own. The timestamp seeking mechanism is the only challenging part.

**Reply by Clemens-E** [reply]
Thanks for the reply!
It was pretty "easy" to do (in 1-2 days).
I'm not knowledgeable in Go, so I decided to write this in Node.js.
My file length is 5 Minutes, so I just list all the files and display it as buttons in HTML, so the user can click on times and start streaming at that position.
When requested, I then use ffmpeg to create all the files for a HLS stream and start streaming, if the stream reaches the end, I jump to the next file (if available) or switch to live view.

Not production ready but a solid prototype, in my opinion

[a comment by an automated account is not shown]
